### PR TITLE
[simix] Refcounting with SIMIX_cond_{ref,unref}()

### PR DESCRIPTION
**Pull request by adfaure**

### CODE REVIEW

File: src/simix/smx_synchro.cpp [hunk 1]
```diff
@@ -356,19 +358,34 @@ void SIMIX_cond_broadcast(smx_cond_t cond)
...
   XBT_DEBUG("Destroy condition %p", cond);
-
   if (cond != nullptr) {
+    intrusive_ptr_release(cond);
+  }
+  XBT_OUT();
+}
+
+
+void intrusive_ptr_add_ref(s_smx_cond_t *cond)
```
Comment (randomstuff) [inline src/simix/smx_synchro.cpp]: I would define those `inline` in the header I guess.

**Comment by randomstuff** [comment]
Looks nice. I'll compile/test that tomorrow before merging it.

File: src/simix/smx_synchro.cpp (hunk 1 above)
Comment (adfaure) [inline src/simix/smx_synchro.cpp]: I would to but the header in which s_smx_cond_t is defined is private which lead to undefined references. Maybe could I put it into simix.h ?

**Comment by randomstuff** [comment]
Merged. Thanks!

**Pull request merged**